code/game/objects/structures/mirror.dm

```diff
@@ -36,44 +73,149 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/mirror, 28)
 
 MAPPING_DIRECTIONAL_HELPERS(/obj/structure/mirror/broken, 28)
 
-/obj/structure/mirror/attack_hand(mob/user, list/modifiers)
+/obj/structure/mirror/attack_hand(mob/living/carbon/human/user)
 	. = ..()
-	if(.)
-		return TRUE
-	if(broken || !Adjacent(user))
-		return TRUE
 
-	if(!ishuman(user))
+	if(. || !ishuman(user) || broken)
 		return TRUE
-	var/mob/living/carbon/human/hairdresser = user
 
-	//handle facial hair (if necessary)
-	if(hairdresser.gender != FEMALE)
-		var/new_style = tgui_input_list(user, "Select a facial hairstyle", "Grooming", GLOB.facial_hairstyles_list)
+	if(!user.can_perform_action(src, FORBID_TELEKINESIS_REACH) && !magical_mirror)
+		return TRUE //no tele-grooming (if nonmagical)
+
+	return what_is_your_wish_masta(user)
+
+/obj/structure/mirror/proc/what_is_your_wish_masta(mob/living/carbon/human/user)
+	var/pick = show_radial_menu(user, src, mirror_options, user, radius = 36, require_near = TRUE)
+	if(!pick)
+		return TRUE //get out
+
+	switch(pick)
+		if(CHANGE_HAIR)
+			change_hair(user)
+		if(CHANGE_BEARD)
+			change_beard(user)
+		if(CHANGE_RACE)
+			change_race(user)
+		if(CHANGE_SEX) // sex: yes
+			change_sex(user)
+		if(CHANGE_NAME)
+			change_name(user)
+		if(CHANGE_EYES)
+			change_eyes(user)
+
+	return what_is_your_wish_masta(user)
+
+/obj/structure/mirror/proc/change_beard(mob/living/carbon/human/beardresser)
+	if(beardresser.physique != FEMALE && !magical_mirror)
+		var/new_style = tgui_input_list(beardresser, "Select a facial hairstyle", "Grooming", GLOB.facial_hairstyles_list)
 		if(isnull(new_style))
 			return TRUE
-		if(!user.can_perform_action(src, FORBID_TELEKINESIS_REACH))
-			return TRUE //no tele-grooming
-		if(HAS_TRAIT(hairdresser, TRAIT_SHAVED))
-			to_chat(hairdresser, span_notice("If only growing back facial hair were that easy for you..."))
+		if(HAS_TRAIT(beardresser, TRAIT_SHAVED))
+			to_chat(beardresser, span_notice("If only growing back facial hair were that easy for you... The reminder makes you feel terrible."))
+			beardresser.add_mood_event("bald_hair_day", /datum/mood_event/bald)
 			return TRUE
-		hairdresser.set_facial_hairstyle(new_style, update = TRUE)
+		beardresser.set_facial_hairstyle(new_style, update = TRUE)
 	else
-		hairdresser.set_facial_hairstyle("Shaved", update = TRUE)
+		if(beardresser.facial_hairstyle == "Shaved")
+			to_chat(beardresser, span_notice("You look towards your facial hair for a make-up and realize you don't have any."))
```

Comment:
what? this is such a confusing feedback message. maybe "You realize you don't have any facial hair!" or something idk